packages/material-ui/src/index.d.ts

```diff
@@ -1,3 +1,4 @@
+/// <reference path="./systemThemeAugmentation.d.ts" />
```

Comment:
Not sure what is the best approach for augmenting the theme inside the `@material-ui/system`. This doesn't seem to work 😕 

What I am trying to do is: If Box is imported from `@material-ui/core/Box`, I'd like the `sx`'s props `theme` to be the MUI Theme. Currently the types are failing - https://app.circleci.com/pipelines/github/mui-org/material-ui/42378/workflows/a8bc65f2-8127-4600-9e4a-a668527fe814/jobs/253818

Reply:
cc @oliviertassinari @eps1lon would be great if you can take a look, I am probably be doing something wrong. Let me know if you need more clarification of what I am trying to achieve.